### PR TITLE
SNOW-1808626 Fix struct type statistic for missing fields

### PR DESCRIPTION
**Pull request by sfc-gh-alhuang**
The original logic didn't calculate null count for missing fields, fix this.

### CODE REVIEW

File: src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergStructuredIT.java [hunk 1]
```diff
@@ -464,4 +471,25 @@ private void assertMap(String dataType, Map<?, ?> value) throws Exception {
     JsonNode expectedNode = value == null ? null : objectMapper.valueToTree(value);
     assertThat(actualNode).isEqualTo(expectedNode);
   }
+
+  private void removeNullFields(JsonNode node) {
```
Comment (sfc-gh-hmadan) [inline src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergStructuredIT.java:475]: add comment on why this method is needed?

Also we should have null value comparison too no?

**Review by sfc-gh-hmadan — APPROVED** [review]

File: src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergStructuredIT.java (hunk 1 above)
Comment (sfc-gh-alhuang) [reply src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergStructuredIT.java:475]: This method is used to compare {} and {key: null}. As this two values are the same for iceberg struct.

Comment (sfc-gh-hmadan) [reply src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergStructuredIT.java:481]: I see. we pass in key=null, but we get back missing-key?

Comment (sfc-gh-alhuang) [reply src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergStructuredIT.java:481]: Yes, both are the same.

**Pull request merged**